image/config.go

```diff
@@ -97,12 +97,21 @@ func (c *config) runtimeSpec(rootfs string) (*specs.Spec, error) {
 
 	var s specs.Spec
 	s.Version = "0.5.0"
+	// we should at least apply the default spec, otherwise this is totally useless
+	s.Process.Terminal = true
 	s.Root.Path = rootfs
-	s.Process.Cwd = c.Config.WorkingDir
-	s.Process.Env = append([]string(nil), c.Config.Env...)
-	s.Process.Args = append([]string(nil), c.Config.Entrypoint...)
+	s.Process.Cwd = "/"
+	if c.Config.WorkingDir != "" {
+		s.Process.Cwd = c.Config.WorkingDir
+	}
+	s.Process.Env = append(s.Process.Env, c.Config.Env...)
+	s.Process.Args = append(s.Process.Args, c.Config.Entrypoint...)
 	s.Process.Args = append(s.Process.Args, c.Config.Cmd...)
 
+	if len(s.Process.Args) == 0 {
+		s.Process.Args = append(s.Process.Args, "sh")
```

Comment:
ocitools is also merging a library which will ease all of this and painlessly merge the image's config and the default one.